tests/Model/Release/ComposerConstraintTest.php

```diff
@@ -120,6 +120,10 @@ public function testParseDev()
         $constraint = new ComposerConstraint('^4.1@dev');
         $this->assertEquals('4.1.0-alpha1', $constraint->getMinVersion()->getValue());
         $this->assertEquals('4.99999.99999', $constraint->getMaxVersion()->getValue());
+
+        $constraint = new ComposerConstraint('4.2.0.x-dev');
+        $this->assertEquals('4.2.0-alpha1', $constraint->getMinVersion()->getValue());
+        $this->assertEquals('4.2.99999', $constraint->getMaxVersion()->getValue());
```

Comment:
this feels wrong to me and the max version should be 4.2.0, however the `ComposerConstraint::parse` method is unable to make any distinction about an explicit `.0` patch version and an implicit one, meaning we have to do 4.2.99999.

I'll run some local tests with cow to see how this works

Reply:
If it helps, composer does something like this too

Reply:
max version should be 4.2.0.999999999. That's how composer internally translates `.x-dev`. It's missing the '0'

Reply:
Yeah, ok - I think @tractorcow is right - there's just no way with our current API to discern if the patch number was explicit or implied :(

Reply:
Maybe look at branching strategy?